DYDX-Bots/Large_USDC_Staking-Withdrawals/src/balance.fetcher.ts

```diff
@@ -0,0 +1,28 @@
+import { providers, Contract, BigNumber } from "ethers";
+import LRU from "lru-cache";
+import { USDC_IFACE } from "./utils";
+
+export default class BalanceFetcher {
+  readonly provider: providers.Provider;
+  private cache: LRU<string, Promise<BigNumber>>;
+  readonly stakeTokenAddress: string;
+  private stakeTokenContract: Contract;
+
+  constructor(provider: providers.Provider, tokenAddr: string) {
+    this.provider = provider;
+    this.cache = new LRU<string, Promise<BigNumber>>({
+      max: 10000,
+    });
+    this.stakeTokenAddress = tokenAddr;
+    this.stakeTokenContract = new Contract(tokenAddr, USDC_IFACE, provider);
+  }
+
+  public async getBalanceOf(address: string, block: string | number): Promise<BigNumber> {
+    const key: string = `${this.stakeTokenContract} - ${address} - ${block}`;
```

Comment:
```suggestion
    const key: string = `${address} - ${block}`;
```
This is enough, the `stakeTokenContract` is always the same, so no need to add it in the key